### PR TITLE
Decouple sitemap load/selection from UI

### PR DESCRIPTION
**Pull request by FlorianSW**
In preparation for #538, it would be easier to have a central code class,
which handles the load of the possible sitemaps without any UI
interaction. This commit changes the current sitemap selection to use such
a class. The UI interaction (select sitemap dialog, e.g.) is still in the
UI class OpenHABMainActivity, the load of the data was moved to the
SitemapList class, which handles the load of the sitemap list.

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/core/sitemap/SitemapList.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+    public ArrayList<OpenHABSitemap> asList() {
+        return new ArrayList<>(sitemaps);
+    }
+
+    public void showSelectSitemapDialog(final Context context, final SitemapSelectCallback cb) {
+        Log.d(TAG, "Opening sitemap selection dialog");
+        final List<String> sitemapNameList = new ArrayList<>();
+        for (int i = 0; i < sitemaps.size(); i++) {
+            sitemapNameList.add(sitemaps.get(i).getName());
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/core/sitemap/SitemapList.java]: getLabel() here

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/core/sitemap/SitemapList.java]: Please look at #597

**Comment by mueller-ma** [comment]
@FlorianSW I remember I reviewed it and some feature from a newer PR was missing, but I cannot see my comment here. I also don't find that missing feature anymore. Strange

**Comment by mueller-ma** [comment]
Found it: #597 is missing here

**Comment by maniac103** [comment]
This class doesn't exactly do what's written on the box, though: as it handles showing the selection dialog, it still includes UI pieces.

I'm not sure whether that decoupling really is that useful, given that fetching the sitemap list doesn't include much logic. IMHO it's more worthwhile cleaning the existing code, e.g. like in this commit: https://github.com/maniac103/openhab.android/commit/9b37f162d1f2fe71f621298ec089d6a067676c77

**Pull request closed without merging**